src/main/java/com/minecolonies/coremod/entity/citizen/EntityCitizen.java

```diff
@@ -399,6 +414,209 @@ public InteractionResult checkAndHandleImportantInteractions(final Player player
         return InteractionResult.SUCCESS;
     }
 
+    /**
+     * Direct interaction actions with a player
+     *
+     * @param player
+     * @param hand
+     * @return interaction result
+     */
+    private InteractionResult directPlayerInteraction(final Player player, final InteractionHand hand)
+    {
+        final ItemStack usedStack = player.getItemInHand(hand);
+
+        if (isInteractionItem(usedStack) && interactionCooldown > 0)
+        {
+            if (!level.isClientSide())
+            {
+                playSound(SoundEvents.VILLAGER_NO, 0.5f, (float) SoundUtils.getRandomPitch(getRandom()));
+                player.sendMessage(new TranslatableComponent("com.minecolonies.coremod.interaction.notnow", this.getCitizenData().getName()).setStyle(Style.EMPTY.withColor(
+                    ChatFormatting.RED)),
+                  player.getUUID());
+            }
+            return null;
+        }
+
+        if (usedStack.getItem() == Items.GOLDEN_APPLE && getCitizenDiseaseHandler().isSick())
```

Comment:
Why is this specifically a golden apple, when that's only half of the items needed to cure one disease (that might not be the one they have)?  Shouldn't it just check if the item is one needed for their current disease's cure, then just add it to inventory (with no cooldown) and use the standard code for curing once they have all of them?

Reply:
yeah this should have a cooldown too.

Reply:
I'd rather make this a more of 1/5th though

Reply:
Yeah but if one of the diseases is bad enough to require a golden apple *and* something else, why can just the apple cure it?  Why would someone try using the apple when the citizen is specifically asking for different items?  Why, since most of those are food items, should the citizen just eat them if given instead of holding on to them to cure?

Perhaps this could use the enchanted golden apple instead, or a healing potion of some sort (then perhaps a high chance of success is ok), but I still think it should also accept the specific healing items too.

Reply:
Atm there is already a small chance of a "miracle healing" in general. The idea was that independent of the disease there is a small chance that this helps enough (kinda realistic compared to reality).

Reply:
Hmm.  Ok, I see why not "enchanted golden apple" now.  I remember those having a crafting recipe, but apparently that was a modpack thing and they don't by default.

I still think it should use a potion rather than golden apple though, and regardless should also detect and accept the proper cure items.  Someone's just going to report it as a bug later otherwise.

Reply:
I can't remember how often I have right clicked a citizen right after I harvested some potatoes and carrots, and still have either of those in my hand.
It would be inconvenient at the very least if the citizen decided to eat the potato or carrot upon interacting, where I actually meant to open their interface to give both items to them.

Reply:
I'll up the chance to just 1 in 3, but the important distinction here is discoverability. There is and should not be any info about this future spread around publically, it is partly a reward of discovering the interaction possibility. Cost wise we're fine with 1.5Apples per cure on avg.
Some annoyance with interact-feeding is expected/factored in, but I'll turn it off for sick people, those realistically aren't that interactive